BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1411,7 +1417,7 @@ public override void OnFixedUpdate()
                     {
                         if (torpedo)
                         {
-                            if (vessel.altitude > 0)
+                            if (vessel.altitude >= -2) //0 was causing airdropped torps to set to 1 crashtolerance, then immediately explode the next frame during splashdown
```

Comment:
Does this depend on the vertical speed of the torpedo? I.e., would a frame or two of `vessel.verticalSpeed` or similar be more appropriate?
Also, the entire `if (TimeIndex > 0.5f)` block might be better served as a coroutine started when the torpedo gets fired that sets the `crashTolerance` to `waterImpactTolerance` after 0.5s, then reverts it to 1 (or whatever it was previously) the frame after it's spashed.
Also also, what happens if it hits land? Is it meant to survive that or is the `crashTolerance` not high enough for that?

Reply:
It really would be better as a one and done coroutine instead of getting continuously checked in FixedUpdate. Legacy Code strikes again. 
waterImpactTolerance should be low enough; testing with the airdropped torps revealed that water impacts are only looking at vertical speed component of impact; in the event is isn't, there's also a if (vessel.landed) Detonate() check at the beginning of boostRoutine to catch any torps that survive the drop.